src/shims/unix/foreign_items.rs

```diff
@@ -295,6 +296,44 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                     }
                 }
             }
+            "aligned_alloc" => {
+                // This is a C11 function, we assume all Unixes have it.
+                // (MSVC explicitly does not support this.)
+                let [align, size] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let align = this.read_target_usize(align)?;
+                let size = this.read_target_usize(size)?;
+
+                // The size must be a multiple of the alignment.
+                // Alignment must be a power of 2, and "supported by the implementation".
+                // If any of these are violated, we have to return NULL.
+                // All fundamental alignments must be supported.
+                // We decide that our implementation supports all alignments.
+                //
+                // macOS and Illumos are buggy in that they require the alignment
+                // to be at least the size of a pointer. We do not emulate those platform bugs.
+                //
+                // Linux also sets errno to EINVAL, but that's non-standard behavior that we do not
+                // emulate/
+                // FreeBSD says some of these cases are UB but that's violating the C standard.
+                // http://en.cppreference.com/w/cpp/memory/c/aligned_alloc
+                // Linux: https://linux.die.net/man/3/aligned_alloc
+                // FreeBSD: https://man.freebsd.org/cgi/man.cgi?query=aligned_alloc&apropos=0&sektion=3&manpath=FreeBSD+9-current&format=html
+                match size.checked_rem(align) {
+                    Some(0) if align.is_power_of_two() => {
+                            let align = this.min_align(align, size);
```

Comment:
You can just use `align.max(this.malloc_align(size))`, no need to write a new function for that